usaspending_api/submissions/management/commands/load_submission.py

```diff
@@ -309,22 +301,67 @@ def handle(self, *args, **options):
         self.logger.info('Acquired award procurement data for ' + str(submission_id) + ', there are ' + str(len(procurement_data)) + ' rows.')
 
         for row in procurement_data:
-            # Yes, I could use Q objects here but for maintainability I broke it out
-            award = Award.objects.filter(piid=row['parent_award_id']).first()
-            if award is None:
-                award = Award.objects.filter(fain=row['parent_award_id']).first()
-            if award is None:
-                award = Award.objects.filter(uri=row['parent_award_id']).first()
-            if award is None:
-                self.logger.error('Could not find an award object with a matching identifier: ' + row['parent_award_id'])
-                continue
+            # Try and find the parent award
+            award = None
+            try:
+                award = find_or_create_summary_award(piid=row.get('piid'), fain=row.get('fain'), uri=row.get('uri'), parent_award_id=row.get('parent_award_id'))
+            except:
+                self.logger.info("Could not find parent award with piid " + row.get('piid', '') + " fain " + row.get('fain', '') + " uri " + row.get('uri', '') + " parent " + row.get('parent_award_id', ''))
 
             procurement_value_map = {
                 "award": award,
-                'submission': submission_attributes
+                'submission': submission_attributes,
+                "action_date": datetime.strptime(row['action_date'], '%Y%m%d')
             }
 
-            procurement = load_data_into_model(Procurement(), row, value_map=procurement_value_map, save=True)
+            load_data_into_model(Procurement(), row, value_map=procurement_value_map, save=True)
+
+
+def get_treasury_appropriation_account_tas_lookup(tas_lookup_id, db_cursor):
+    # Checks the broker DB tas_lookup table for the tas_id and returns the matching TAS object in the datastore
+    db_cursor.execute('SELECT * FROM tas_lookup WHERE tas_id = %s', [tas_lookup_id])
+    tas_data = dictfetchall(db_cursor)
+
+    # These or "" convert from none to a blank string, which is how the TAS table stores nulls
+    q_kwargs = {
```

Comment:
Nice, glad to see we're using the broker's TAS table here. Do we need a story to think through the scenarios being discussed in the channel about how TAS accounts can change over time? I wasn't following closely and want to make sure we've captured the info somewhere if there's more work to do.